### ISSUE TITLE
Nullable fields not communicated in types

### ISSUE THREAD

**Issue by meme** [microsoftgraph/msgraph-typescript-typings#148]
Fields in the Graph API can often times be `null`, an example being the result from the "Create User" endpoint:

```js
{
    "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#users/$entity",
    "id": "id-value",
    "businessPhones": [],
    "displayName": "displayName-value",
    "givenName": null,
    "jobTitle": null,
    "mail": null,
    "mobilePhone": null,
    "officeLocation": null,
    "preferredLanguage": null,
    "surname": null,
    "userPrincipalName": "upn-value@tenant-value.onmicrosoft.com"
}
```
(https://docs.microsoft.com/en-us/graph/api/user-post-users?view=graph-rest-1.0&tabs=http#response-1.)

However, the respective typing for `MicrosoftGraph.User`'s `jobTitle` states that it is either `string | undefined` using the `?` TS shorthand:

```ts
    jobTitle?: string;
```
(https://github.com/microsoftgraph/msgraph-typescript-typings/blob/master/microsoft-graph.d.ts#L1235.)

The response from the API (both on the docs and in practice) shows that `jobTitle` can be nullable (as can other fields). The typings should be updated to reflect nullable fields.

**Comment by zengin** [comment]
Yes, our definition is using optional parameter types, not nullable types, so it doesn't reflect what is intended in the metadata.

There are two main ways to fix it, either by defining `Nullable<T>` as `T | null` and consuming it as `Nullable<string>` or defining each type explicitly, e.g. `string | null`. Note that these have the potential of breaking dot notation auto-complete in IDEs.

**Comment by zengin** [comment]
Typescript documentation:
- [Nullable types](https://www.typescriptlang.org/docs/handbook/advanced-types.html#nullable-types)
- [Optional property types](https://www.typescriptlang.org/docs/handbook/advanced-types.html#nullable-types)
